checker/tests/run-pass/smt_shift_left.rs

```diff
@@ -12,17 +12,18 @@ pub fn read_uleb128_as_u32(bytes: [u8; 20]) -> u32 {
     let mut value: u32 = 0;
     let mut shift: u32 = 0;
     let mut cursor = 0;
-    while cursor < bytes.len() {
+    while cursor < bytes.len() && shift <= 28 {
         let byte = bytes[cursor];
         let val = byte & 0x7f;
         value |= (val as u32) << shift;
         if val == byte {
             return value;
         }
         shift += 7;
-        if shift > 28 {
-            break;
-        }
+        // todo: this guard is not treated as the loop's entry condition because it is dominated by the loop anchor
```

Comment:
I think what needs to happen here is that shift must get narrowed in the state that feeds back into the loop anchor.

Reply:
Updated the todo comment.